internal/envoy/manager/envoy_callbacks.go

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) 2022 Kubeshop
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Callbacks are called by GRPC server on new events.
+package manager
+
+import (
+	"context"
+
+	envoy_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/go-logr/logr"
+)
+
+type Callbacks struct {
+	cacheManager *cacheManager
+	logger       logr.Logger
+}
+
+func NewCallbacks(cacheManager *cacheManager, logger logr.Logger) *Callbacks {
+	return &Callbacks{
+		cacheManager: cacheManager,
+		logger:       logger.WithName("CacheManager"),
+	}
+}
+
+func (c *Callbacks) OnStreamOpen(ctx context.Context, id int64, typeUrl string) error {
+	c.logger.Info("OnStreamOpen", "id", id, "typeUrl", typeUrl)
+	return nil
+}
+func (c *Callbacks) OnStreamClosed(id int64) {
+	c.logger.Info("OnStreamClosed", "id", id)
+}
+
+func (c *Callbacks) OnDeltaStreamOpen(ctx context.Context, id int64, typeUrl string) error {
+	c.logger.Info("OnDeltaStreamOpen", "id", id, "typeUrl", typeUrl)
+	return nil
+}
+
+func (c *Callbacks) OnDeltaStreamClosed(id int64) {
+	// `l.logger.V(1)` is effectively debug level.
+	c.logger.Info("OnDeltaStreamClosed", "id", id)
+}
+
+func (c *Callbacks) OnStreamRequest(id int64, request *envoy_discovery_v3.DiscoveryRequest) error {
+	c.logger.Info("OnStreamRequest", "id", id, "request.TypeUrl", request.TypeUrl)
+	if c.cacheManager.IsNodeExist(request.Node.Id) {
+		return nil
+	}
+
+	if err := c.cacheManager.setNodeSnapshot(request.Node.Id, request.Node.Cluster); err != nil {
+		c.logger.Error(err, "OnStreamRequest", "id", id, "request.TypeUrl", request.TypeUrl)
+		return err
+	}
+
+	return nil
+}
+
+func (c *Callbacks) OnStreamResponse(ctx context.Context, id int64, request *envoy_discovery_v3.DiscoveryRequest, response *envoy_discovery_v3.DiscoveryResponse) {
+	c.logger.Info("OnStreamResponse", "id", id, "request.TypeUrl", request.TypeUrl, "response.TypeUrl", response.TypeUrl)
```

Comment:
Do we really want to be flooding the logs with these events if we are not doing anything with them? 
Would it be better to just `return`?

Reply:
I think it's quite useful. You could always filter out the logging by using `grep -v OnStreamResponse` etc, but it's not possible to put the logging back in without writing code.

Make the call, do you want me to remove it or shall I leave it there (my preference), @kylehodgetts

Reply:
We need to think about this from a user perspective. If they are checking the manager logs and they see this, does it help them?

Reply:
I've turned them into debug level log statements then. Please mark as resolved, if you're OK with that.